### ISSUE TITLE
batched-wav-nnet3-cuda-online fail when gpu-feature-extract is true

### ISSUE THREAD

**Issue by housebaby** [kaldi-asr/kaldi#4166]
I met a problem when using batched-wav-nnet3-cuda-online with default  gpu-feature-extract.
```
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online --feature-type=mfcc --mfcc-config=/home/work/release/online/20190408-cell//conf/mfcc.conf --ivector-extraction-config=/home/work/release/online/20190408-cell//conf/ivector_extractor.conf /home/work/release/online/20190408-cell//final.mdl /home/work/release/online/20190408-cell//graph/HCLG.fst scp:/home/work/work/lm/product/cellphone/data/test/mobilephone-2000-20180830_cn/wav.scp 'ark:|gzip -c > ./lat.1.gz' 
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:SelectGpuId():cu-device.cc:223) CUDA setup operating under Compute Exclusive Mode.
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:FinalizeActiveGpu():cu-device.cc:308) The active GPU is [0]: Tesla V100-PCIE-32GB	free:31959M, used:520M, total:32480M, free/total:0.983962 version 7.0
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:RemoveOrphanNodes():nnet-nnet.cc:948) Removed 1 orphan nodes.
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:RemoveOrphanComponents():nnet-nnet.cc:847) Removing 2 orphan components.
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:Collapse():nnet-utils.cc:1472) Added 1 components, removed 2
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:204) Done.
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:204) Done.
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:204) Done.
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:ComputeDerivedVars():ivector-extractor.cc:204) Done.
Loading eval dataset...done
ASSERTION_FAILED (batched-wav-nnet3-cuda-online[5.5.697~1-79790]:AddMatMat():cu-matrix.cc:1317) Assertion failed: (k == k1)

[ Stack-Trace: ]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::MessageLogger::LogMessage() const+0x8b7) [0x96872d]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::KaldiAssertFailure_(char const*, char const*, int, char const*)+0x59) [0x969208]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::CuMatrixBase<float>::AddMatMat(float, kaldi::CuMatrixBase<float> const&, kaldi::MatrixTransposeType, kaldi::CuMatrixBase<float> const&, kaldi::MatrixTransposeType, float)+0xeb) [0x73f62b]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::BatchedIvectorExtractorCuda::LDATransform(kaldi::CuMatrix<float> const&, kaldi::CuMatrix<float>*, kaldi::LaneDesc const*, int)+0x76) [0x60bc6c]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::BatchedIvectorExtractorCuda::GetIvectors(kaldi::CuMatrixBase<float> const&, kaldi::CuVectorBase<float>*, kaldi::LaneDesc const*, int)+0x9b) [0x60c74b]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::OnlineBatchedFeaturePipelineCuda::ComputeFeaturesBatched(int, std::vector<int, std::allocator<int> > const&, std::vector<int, std::allocator<int> > const&, std::vector<bool, std::allocator<bool> > const&, std::vector<bool, std::allocator<bool> > const&, float, kaldi::CuMatrixBase<float> const&, kaldi::CuMatrix<float>*, kaldi::CuVector<float>*, std::vector<int, std::allocator<int> >*)+0x3d0) [0x604ed0]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::cuda_decoder::BatchedThreadedNnet3CudaOnlinePipeline::ComputeGPUFeatureExtraction(std::vector<int, std::allocator<int> > const&, std::vector<kaldi::SubVector<float>, std::allocator<kaldi::SubVector<float> > > const&, std::vector<bool, std::allocator<bool> > const&, std::vector<bool, std::allocator<bool> > const&)+0x246) [0x5d72b6]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(kaldi::cuda_decoder::BatchedThreadedNnet3CudaOnlinePipeline::DecodeBatch(std::vector<unsigned long, std::allocator<unsigned long> > const&, std::vector<kaldi::SubVector<float>, std::allocator<kaldi::SubVector<float> > > const&, std::vector<bool, std::allocator<bool> > const&, std::vector<bool, std::allocator<bool> > const&)+0x103) [0x5d8a65]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online(main+0x16ed) [0x5c85ac]
/lib64/libc.so.6(__libc_start_main+0xf5) [0x7f83fbef3b35]
/home/work/tools/kaldi-master/src/cudadecoderbin/batched-wav-nnet3-cuda-online() [0x5c5f79]

local/test_batch_gpu.sh: line 27: 258145 Aborted                 $EXE --feature-type=mfcc --mfcc-config=${MODEL}/conf/mfcc.conf --ivector-extraction-config=${MODEL}/conf/ivector_extractor.conf ${NNET} ${FST} ${WAV} "$LAT"
```

But if I set  gpu-feature-extract to  false, it works well.
@hugovbraun

**Comment by danpovey** [comment]
Think this was fixed.  Is your repo fully up to date + compiled?

**Comment by kkm000** [comment]
You have a pretty old version [5.5.697~1-79790]
```
* 797905b01 200520 [src] Fix directory in #include statement (#4079)
```
This is 2002-05-20, 2 months ago.

**Comment by kkm000** [comment]
I am closing this ticket for now. If you believe that your issue has not been addressed, please feel free to ping me, and I'll reopen it.

**Comment by housebaby** [comment]
> You have a pretty old version [5.5.697~1-79790]
> 
> ```
> * 797905b01 200520 [src] Fix directory in #include statement (#4079)
> ```
> 
> This is 2002-05-20, 2 months ago.

Yes, you are right. Thanks a lot.
